Sources/MongoSwift/Credential.swift

```diff
@@ -1,5 +1,5 @@
 /// Represents an authentication credential.
-internal struct Credential: Decodable, Equatable {
+public struct Credential: Decodable, Equatable {
```

Comment:
the spec says we SHOULD name this `MongoCredential`: https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#mongocredential

also, this type needs a public initializer now, and let's make all of its properties `public` and `var`s.

that said we did recently decide to move away from having a nested struct for TLS options and instead flattened it into the `ClientOptions` to match the connection string structure. So another option would be to nest this info directly in `ClientOptions` though that does go against the spec.

Opinions?

Reply:
Quick tour of the drivers shows how stretched should really is 😅 :
- [go](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/auth/cred.go#L10)
- [ruby](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/auth/user.rb#L23)
- [rust](https://github.com/mongodb/mongo-rust-driver/blob/master/src/client/auth/mod.rs#L181)
- [javascript](https://github.com/mongodb/node-mongodb-native/blob/master/lib/cmap/auth/mongo_credentials.js#L35)

I think that it does warrant its own sub-struct as it is a neat unit of information constrained to auth only. Additionally it does match other drivers in that way. But I defer to your call on that since tls opts were flattened

Reply:
ok, I'm cool with keeping a separate type then. it sounds like we've reached consensus on keeping the `Mongo` prefixes. so I'll resolve this thread.